rotkehlchen/tests/api/test_solana_transactions.py

```diff
@@ -0,0 +1,74 @@
+@pytest.mark.parametrize('solana_accounts', [['7T8ckKtdc5DH7ACS5AnCny7rVXYJPEsaAbdBri1FhPxY']])
+def test_query_solana_transactions(
+        rotkehlchen_api_server: 'APIServer',
+        solana_accounts: list[SolanaAddress],
+) -> None:
+    """Test that solana transactions are properly queried via the api.
+    Also checks that querying with existing txs in the DB only queries new transactions.
+    """
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    for signatures_list, tx_count in ((  # Make two api calls to test querying with existing txs in the DB.  # noqa: E501
+        (Signature.from_string('5vBFfTGrcdkE7ZdsUDSU2kRkhoFp4EgKtLLB6h2m1uQoG5wCddCkFGnNjXaHrV2r1kZ8CpJfh7UcWJ9tFXAyKc8Q'),
+        Signature.from_string('2ZYFMzQMpDFcAmXo2UMhGErSitNpuZ4zeu548QvMU8k37cgetF91wTYnGmN1oZq6EG7zXaZyNPCzWtakDnSJEtgF')),
+        2,
+    ), (
+        (Signature.from_string('LLco7QQYo9HVc8w6YZeabxrdhZAjQxGRvrk1hNCJPrHGYAELjh3HwQKvTA1n8bWmkcLyKkFivieooK8C9LvYZuy'),),
+        3,
+    )):
+        with (
+            patch.object(
+                target=rotki.chains_aggregator.solana,
+                attribute='query_tx_signatures_for_address',
+                side_effect=lambda address, until, sigs=signatures_list: sigs,
+            ),
+            patch.object(
+                target=rotki.chains_aggregator.solana.helius,
+                attribute='get_transactions',
+                wraps=rotki.chains_aggregator.solana.helius.get_transactions,
+            ) as mock_helius_get_transactions,
+            patch.object(
+                target=rotki.chains_aggregator.solana,
+                attribute='query_rpc_for_single_tx',
+                wraps=rotki.chains_aggregator.solana.query_rpc_for_single_tx,
+            ) as mock_rpc_get_transaction,
+        ):
+            response = requests.post(
+                api_url_for(rotkehlchen_api_server, 'blockchaintransactionsresource'),
+                json={
+                    'accounts': [{'address': solana_accounts[0], 'blockchain': 'solana'}],
+                    'async_query': (async_query := random.choice([False, True])),
+                },
+            )
+            assert assert_proper_response_with_result(response, rotkehlchen_api_server, async_query)  # noqa: E501
+
+        assert mock_helius_get_transactions.call_count == 1  # tries to query helius first (but has no api key)  # noqa: E501
+        assert mock_rpc_get_transaction.call_count == len(signatures_list)  # falls back to doing a single rpc query for each tx  # noqa: E501
+
+        with rotki.data.db.conn.read_ctx() as cursor:
+            transactions = DBSolanaTx(rotki.data.db).get_solana_transactions(
+                cursor=cursor,
+                filter_=SolanaTransactionsFilterQuery.make(),
+            )
+
+        assert len(transactions) == tx_count
+        signatures_from_db = [x.signature for x in transactions]
+        assert all(x.to_bytes() in signatures_from_db for x in signatures_list)
```

Comment:
you can do this instead.
```suggestion
        assert all(x.signature in signature_list for x in transactions)
```